src/Domain/Structure/NeighborIsConforming.hpp

```diff
@@ -0,0 +1,39 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+
+/// \cond
+template <size_t VolumeDim>
+class Direction;
+template <size_t VolumeDim>
+class OrientationMap;
+namespace domain {
+enum class Topology : uint8_t;
+}  // namespace domain
+/// \endcond
+
+namespace domain {
+/// \ingroup ComputationalDomainGroup
+/// \brief Returns whether or not neighboring Blocks (or Elements) have
+/// conforming block logical coordinates on their interface
+///
+/// \details Block logical coordinates are considered to be conforming if they
+/// are identical or related by a discrete rotation (i.e. a valid non-aligned
+/// OrientationMap).  It is a requirement that neighboring Blocks be conforming
+/// if their oriented topologies are the same in the interface dimensions.
+///
+/// \note If the neighbors are conforming, they can exchange boundary data via
+/// copy or projection, taking into account the discrete rotation if necessary.
+/// If the neighbors are not conforming, boundary data will need to be
+/// interpolated.
+template <size_t VolumeDim>
+bool neighbor_is_conforming(
+    const std::array<Topology, VolumeDim>& self_topologies,
+    const std::array<Topology, VolumeDim>& neighbor_topologies,
+    const Direction<VolumeDim>& direction_to_neighbor,
+    const OrientationMap<VolumeDim>& orientation_of_neighbor);
```

Comment:
This seems to ignore refinement. I think that should be stated explicitly

Reply:
is_conforming is with respect to the block, so refinement is irrelevant

Reply:
The documentation literally says " (or Elements)"

Reply:
I agree.  But it also says that it is the block logical coordinates (of the Block or Element) which define if the neighbors are conforming...   So please suggest what wording would clarify the documentation for you...

Reply:
Either remove "(or Elements)" or add something like "\note This does not take into account h or p refinement between neighbor elements, so projection may still be necessary."

Reply:
Okay after discussion, will update to 
```
/// \note If neighboring Elements are conforming, they can exchange boundary data via
/// either copy (if they have the same h- and p-refinement) or projection (if they
/// don't), taking into account the discrete rotation if necessary.
/// If the neighbors are not conforming, boundary data will need to be
/// interpolated.
```